api/kyverno/v1/image_verification_types.go

```diff
@@ -226,9 +226,23 @@ type KeylessAttestor struct {
 
 type CTLog struct {
 	// URL is the address of the transparency log. Defaults to the public log https://rekor.sigstore.dev.
-	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:Default:=https://rekor.sigstore.dev
 	URL string `json:"url" yaml:"url"`
+
+	// RekorPubKey is an optional PEM encoded public key to use for a custom Rekor.
+	// +kubebuilder:validation:Optional
+	RekorPubKey string `json:"pubkey,omitempty" yaml:"pubkey,omitempty"`
```

Comment:
Previously, one use case we supported was to verify a keyless attestation or image signature using the root certificate of Rekor.

Will this still be supported? May be good to add a `kutt` test case for this.

Reply:
In the current behavior, we are providing the public rekor url and its public key by default, to test `RekorPubKey` with a kuttl test, I believe we will have to setup an instance of rekor for test.

So, I created this unit test (https://github.com/kyverno/kyverno/pull/7248/commits/497646a5c03766ef5805fcab54862c0451f7071a) where we:
1. Pass an invalid rekor url so it doesn't use the default rekor URL.
2. Pass an invalid rekor public key so that it throws an error.
3. Pass the right rekor public key and the test passes.

It tests that we can verify signatures only using the rekor public key

Reply:
That is not the same. The requirement was to allow the user to not specify Rekor and only provide a root CA.

Reply:
Can they specify `ignoreTlog` to achieve this?

Reply:
> That is not the same. The requirement was to allow the user to not specify Rekor and only provide a root CA.

@JimBugwadia I am not sure if I completely understand this.

- If a user want to verify just the signature and not rekor entry, then they can set ignoreTlogs  for that
- If the user has the rekor root CA, they can just provide that CA and not the rekor url, and cosign will do the verification based on that rekor URL

Assuming that the root CA mentioned by you is the Rekor Root CA, for a keyless verification, the user can provide
1. Identity and Issuer
2. Rekor Root CA

This will be enough to verify the signature and the tlog